core/base/src/TObject.cxx

```diff
@@ -284,7 +284,10 @@ void TObject::Execute(const char *method, const char *params, Int_t *error)
 
    gInterpreter->Execute(this, IsA(), method, params, error);
 
-   if (gPad && must_cleanup) gPad->Modified();
+   if (gPad && must_cleanup) {
+      gPad->Modified();
+      gPad->Update();
```

Comment:
This seems unrelated to the PR and should at least be its own commit.   

Also this is supposed to be redundant and the "event loop" is suppose to be calling Update as needed (when Modified is set/noticed).

Reply:
OK, no problem, I'll remove it (but on Windows the canvas is not updated...)

Reply:
Yes but that change simply 'hide' the problem, in my humble opinion :)

Reply:
Fair enough